connection_go18.go

```diff
@@ -63,9 +56,25 @@ func (mc *mysqlConn) BeginTx(ctx context.Context, opts driver.TxOptions) (driver
 		}
 	}
 
+	if opts.ReadOnly {
+		return mc.beginReadOnly()
+	}
 	return mc.Begin()
 }
 
+func (mc *mysqlConn) beginReadOnly() (driver.Tx, error) {
```

Comment:
just add a parameter to `mc.Begin` and reuse it.
The implementation is already inconsistent between the two functions.